code/game/objects/effects/decals/crayon.dm

```diff
@@ -3,29 +3,205 @@
 	desc = "A rune drawn in crayon."
 	icon = 'icons/obj/rune.dmi'
 	layer = TURF_DECAL_LAYER
-	anchored = 1
+	anchored = TRUE
 	random_rotation = 0
+	var/is_rune = FALSE
 
-	New(location,main = "#FFFFFF",shade = "#000000",var/type = "rune")
-		..()
-		loc = location
+/obj/effect/decal/cleanable/crayon/New(location,main = "#FFFFFF",shade = "#000000",var/type = "rune")
+	..()
+	loc = location
 
-		name = type
-		desc = "A [type] drawn in crayon."
+	name = type
+	desc = "A [type] drawn in crayon."
 
-		switch(type)
-			if("rune")
-				type = "rune[rand(1,6)]"
-			if("graffiti")
-				type = pick("amyjon","face","matt","revolution","engie","guy","end","dwarf","uboa")
+	switch(type)
+		if("rune")
+			type = "rune[rand(1,6)]"
+			is_rune = TRUE
+		if("graffiti")
+			type = pick("amyjon","face","matt","revolution","engie","guy","end","dwarf","uboa")
 
-		var/icon/mainOverlay = new/icon('icons/effects/crayondecal.dmi',"[type]",2.1)
-		var/icon/shadeOverlay = new/icon('icons/effects/crayondecal.dmi',"[type]s",2.1)
+	var/icon/mainOverlay = new/icon('icons/effects/crayondecal.dmi',"[type]",2.1)
+	var/icon/shadeOverlay = new/icon('icons/effects/crayondecal.dmi',"[type]s",2.1)
 
-		mainOverlay.Blend(main,ICON_ADD)
-		shadeOverlay.Blend(shade,ICON_ADD)
+	mainOverlay.Blend(main,ICON_ADD)
+	shadeOverlay.Blend(shade,ICON_ADD)
 
-		add_overlay(mainOverlay)
-		add_overlay(shadeOverlay)
+	add_overlay(mainOverlay)
+	add_overlay(shadeOverlay)
 
-		add_hiddenprint(usr)
\ No newline at end of file
+	add_hiddenprint(usr)
+
+/obj/effect/decal/cleanable/crayon/attackby(obj/item/I, mob/living/carbon/human/M)
+	..()
+	if(istype(I, /obj/item/oddity/common/book_unholy) || istype(I, /obj/item/oddity/common/book_omega))
+		if(M.disabilities&NEARSIGHTED && is_rune && M.species?.reagent_tag != IS_SYNTHETIC)
+
+			//Anti-Death check
+			if(M.maxHealth <= 30)
+				to_chat(M, "<span class='info'>You try to do as the book wrights but the failing body prevents you form even mubbling a single more word.</span>")
+				return
+
+			var/obj/item/flame/candle/mage_candle
+			var/datum/reagent/organic/blood/B = M.get_blood()
+			var/obj/effect/decal/cleanable/blood/writing/spell
```

Comment:
Why is spell & mage_candle defined here if they are only used for the for() loop.